### PR TITLE
Replace `<SelectIDE />` with `<SelectIDEComponent />`

### LINKED ISSUE #13116: Replace the editor selection cards with the new dropdown component used in the start with options new workspace modal
### Problem to solve

Currently, we're using the [`SelectableCardSolid`](https://github.com/gitpod-io/gitpod/blob/fe26d2613df2f85667829c9de86bbeb187dc81cf/components/dashboard/src/components/SelectableCardSolid.tsx) component in multiple areas in the dashboard like the Editor, Workspaces, etc.

While the options for _Editor_ and _Workspaces_ will grow over time, they will always remain finite. See https://github.com/gitpod-io/gitpod/issues/7751 and https://github.com/gitpod-io/gitpod/issues/12708.

However, this card component can not scale well after adding too many options, especially when the card will have to contain multiple rows of information like the _Workspaces_ (class) selection.

See also [relevant discussion](https://gitpod.slack.com/archives/C01KPEPQYE6/p1663594639896499) (internal). Cc @loujaybee 

### Proposal

~~Introduce a new, more flexible component or pattern like a dropdown that is able to scale better for selecting from a range of options like the _Editor_ and _Workspaces_.`~~

Replace the editor selection cards with the new dropdown component used in the start with options new workspace modal.

See [relevant discussion](https://gitpod.slack.com/archives/C01KPEPQYE6/p1672836161790069) (internal). Cc @akosyakov 

| BEFORE / AFTER |
|-|
| editor |

### PR DESCRIPTION
**Pull request by filiptronicek**
## Description
Replaces the IDE selection cards with the dropdown adopted in contexts like the user onboarding and the new workspace modal. 

This also removes the onboarding "Select IDE" modal, because that is now a part of the new onboarding flow.

## Related Issue(s)

Fixes #13116 

## How to test
Take a look at it at https://filiptroni9852fbf578.preview.gitpod-dev.com/user/preferences

## Release Notes

```release-note
NONE
```

| Before | After |
|--------|--------|
| image | image | 

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] with-ee-license
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by gtsiolis** [comment]
Looking at this now! 👀

File: components/dashboard/src/user-settings/SelectIDE.tsx
Comment (gtsiolis) [inline components/dashboard/src/user-settings/SelectIDE.tsx:80]: **issue**: Probably not the best line of code to comment on this, but could we make sure the editor dropdown renders above the modal (`z-index`)? This is an issue regardless if we change the modal width in the scope of this PR.

| Issue on current modal width | Issue on changed modal width |
|-|-|
| Screenshot 2023-03-16 at 13 59 56 | Screenshot 2023-03-16 at 13 54 39 |

**Review by gtsiolis — APPROVED** [review]
Thanks for making this change, @filiptronicek! 🌮 🌮 

Left two minor issue comments we can tackle also in a separate PR.

Approving to unblock merging, but holding in case you'd like to address the issue comments in this PR. 🏓 

/hold

File: components/dashboard/src/user-settings/SelectIDE.tsx
Comment (filiptronicek) [reply components/dashboard/src/user-settings/SelectIDE.tsx:80]: @gtsiolis how did you trigger the modal? I can't seem to be able to get to it from the UI

Comment (gtsiolis) [reply components/dashboard/src/user-settings/SelectIDE.tsx:80]: @filiptronicek We show this modal to all new users—you can reproduce if you delete your account and try to login again. 💡

Comment (filiptronicek) [reply components/dashboard/src/user-settings/SelectIDE.tsx:80]: @gtsiolis I think I got it a couple of times, but it's always under the new onboarding experience and when I finish that, it's gone.

Comment (gtsiolis) [reply components/dashboard/src/user-settings/SelectIDE.tsx:80]: @filiptronicek The new onboarding flow is not available on preview environments, see [relevant discussion](https://gitpod.slack.com/archives/C04PH56LZK5/p1678105071868449) (internal).

❓ Are you seeing the new onboarding flow in the preview environment for this PR? 💭 

💡 I could reliably re-trigger this modal after deleting my account quite a few times in this preview environment. Steps top reproduce:
1. Delete the account from [`/user/account`](https://filiptroni9852fbf578.preview.gitpod-dev.com/user/account) in the preview environment.
2. Try to logging in again.

Comment (filiptronicek) [reply components/dashboard/src/user-settings/SelectIDE.tsx:80]: > ❓ Are you seeing the new onboarding flow in the preview environment for this PR? 💭 

Yes

Comment (gtsiolis) [reply components/dashboard/src/user-settings/SelectIDE.tsx:80]: That's strange—could you share a screenshot of what you're seeing and what URL you're using?

Comment (filiptronicek) [reply components/dashboard/src/user-settings/SelectIDE.tsx:80]: Aaah 🤦, I was explicitly mention in the feature flag to be included. Thanks for the help.

Comment (gtsiolis) [reply components/dashboard/src/user-settings/SelectIDE.tsx:80]: Resolving this as the modal width now seems better! 📕

Comment (gtsiolis) [reply components/dashboard/src/user-settings/SelectIDE.tsx:80]: The modal width now seems better now! 📕

Comment (filiptronicek) [reply components/dashboard/src/user-settings/SelectIDE.tsx:80]: https://github.com/gitpod-io/gitpod/pull/16847/commits/9610c3679ed83a4ead08244b78e3f7cb425982b7 ✅

Comment (filiptronicek) [reply components/dashboard/src/user-settings/SelectIDE.tsx:80]: I have played around with the positioning of the dropdown. I have tried many different tricks with positioning and setting the `z-index`, but only one of those attempts actually worked out.

We apparently need to use `position: fixed` for the position (currently, `position: absolute` is used), which is problematic because of its impact of relative positioning -> when positioning a `fixed` element, I could not get the dropdown options to be relative to the dropdown component, but that should be intended `fixed` behavior [[MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/position#values)]. The problem is that now the dropdown options are flying on the top of the viewport and the only way to get them to their proper position is to manually try out different `top` values, until one is seemingly correct[^1].

image

[^1]: which is obviously problematic, because it depends on the height of the content above the dropdown, so that if you change the "Choose the editor for opening workspaces" text and make it a bit longer, the `top` value will need to be re-adjusted.

Comment (filiptronicek) [reply components/dashboard/src/user-settings/SelectIDE.tsx:80]: cc @selfcontained do you know of anything that might help here?

Comment (gtsiolis) [reply components/dashboard/src/user-settings/SelectIDE.tsx:80]: @filiptronicek Here's a somewhat radical idea that could help: Last time I checked there was consensus to drop this editor selection and replace it with the new onboarding flow. Given that we're already doing this on Gitpod.io, we could drop this all together which can help bypass this issue. Later on, we could think about reusing the onboarding for Dedicated, which already includes this editor selection in the step 2. 

@loujaybee What do you think? Dropping this could help as it's only our team who sees this now in the preview environments. 🏓

Comment (gtsiolis) [reply components/dashboard/src/user-settings/SelectIDE.tsx:80]: @filiptronicek We've got green flag to remove this, see [relevant discussion](https://gitpod.slack.com/archives/C01KGM9BH54/p1678996938846779?thread_ts=1678995547.356729&cid=C01KGM9BH54) (internal). Cc @loujaybee 

Could you make this change in this PR? Otherwise, a separate PR sounds also fine—your call. 🏓

**Review by gtsiolis — APPROVED** [review]
UX LGTM! 🌮 🌮

File: components/dashboard/src/user-settings/Preferences.tsx
Comment (selfcontained) [inline components/dashboard/src/user-settings/Preferences.tsx:23]: This looks like it's not used anywhere. Was it copied over from `SelectIDE.tsx` during dev and then forgot to remove maybe?

Not something we need to do here, but figured I'd comment for context: FWIW for any new user-related updates, I'd love to start moving calls to use react-query mutations. We've got one that we setup w/ the new onboarding flow that can be used the same way. ATM we still have to update UserContext, but we can move away from that as we swap more things to react-query for the user. https://github.com/gitpod-io/gitpod/blob/3f8e507cb7dccc0b259f3152669390bdd41dd7e7/components/dashboard/src/onboarding/UserOnboarding.tsx#L38.

**Review by selfcontained — APPROVED** [review]
One comment about some code I think we can remove. LGTM though, and cleans it up nicely. Thanks!

/hold in case you want to make the change, feel free to do it in a followup too though.

**Pull request merged**
